src/schemes/cookie.ts

```diff
@@ -16,8 +18,20 @@ const DEFAULTS = {
 }
 
 export default class CookieScheme extends LocalScheme {
+  public req
+
   constructor ($auth, options) {
     super($auth, options, DEFAULTS)
+
+    this.req = $auth.ctx.req
+  }
+
+  mounted () {
+    if (process.server) {
+      this.$auth.ctx.$axios.setHeader('referer', requrl(this.req))
```

Comment:
Can we directly use `this.$authx.ctx.req` here? Also might be a good idea linking issue with a comment on why we would need to set referrer.

Reply:
One more question: Is refer check only by hostname or full path?

Reply:
Ok

Reply:
I think it's only hostname. At least it woked for me and @radudiaconu0